### PR TITLE
Displaying adequate fields in hex

### LINKED ISSUE #644: Minor breaking change after the 0.3.1 release
## Protostar version

| Name       | Version              |
| ---------- | -------------------- |
| Protostar  | `0.3.1^`  |
| Cairo-lang | `0.9.1` |
| Git        | `2.34.1`        |

## Current Behavior
After a successful deployment, the ouput of the tx hash is now a felt.
It is not a major issue but since starknet API `https://alpha4.starknet.io/feeder_gateway/get_transaction_status?transactionHash=` expects a hex, it is less conveniant.

```
Deploy transaction was sent.
Contract address: 0x02c4672468cd6116ae2d8dc07b8d12e2561593e392fa2e693ac4b8576ecaab20
Transaction hash: 3379530798764605818090729322813257897802304273379266993039341877985495062261
```

## Expected Behavior
In previous versions it was a hex (note I didn't test to install a previous version, but some of our scripts does not work anyone due to this)

```
Deploy transaction was sent.
Contract address: 0x02c4672468cd6116ae2d8dc07b8d12e2561593e392fa2e693ac4b8576ecaab20
Transaction hash: 0x778BF24408D10F7262177110CADB96F3CE6E5556D9BB4A8BBBC807D76D196F5
```

## Steps to Reproduce
Deploy a contract.

## Possible Solution
I just had to add a `felt_to_hex` parser to fix it locally.

### PR DESCRIPTION
**Pull request by Radinyn**
fix #644

### CODE REVIEW

**Review by mkaput — APPROVED** [review]

**Review by kasperski95 — COMMENTED** [review]
There should be a test checking transaction hash representation.

File: protostar/starknet_gateway/gateway_response_test.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    assert formatted[0] == "Declare transaction was sent."
+
+    class_hash = formatted[1][formatted[1].find(":") + 1 :].strip()
+    transaction_hash = formatted[2][formatted[2].find(":") + 1 :].strip()
+
+    assert class_hash == f"0x{test_input.class_hash:064x}"
+    assert int(class_hash, 16) == test_input.class_hash
+
+    assert transaction_hash == f"0x{test_input.transaction_hash:064x}"
+    assert int(transaction_hash, 16) == test_input.transaction_hash
```
Comment (mkaput) [inline protostar/starknet_gateway/gateway_response_test.py:51]: wdyt about also including an assertion for this in e2e tests somewhere? as a smoke test that this logic is properly integrated

**Review by kasperski95 — APPROVED** [review]

File: protostar/starknet_gateway/starknet_request.py [hunk 2]
```diff
@@ -55,8 +55,14 @@ def prettify_payload(
             if len(key) > max_key_length:
                 max_key_length = len(key)
 
+        # Will be converted to hex
+        as_hex = {"transaction_hash", "contract_address", "class_hash"}
+
         first_column_width = max(max_key_length, 20)
         for key, value in payload.items():
+            if key in as_hex:
+                value = f"0x{value:064x}"
```
Comment (mkaput) [inline protostar/starknet_gateway/starknet_request.py:64]: plz add a test for this

**Review by mkaput — CHANGES_REQUESTED** [review]

File: protostar/starknet_gateway/starknet_request.py (hunk 2 above)
Comment (Radinyn) [reply protostar/starknet_gateway/starknet_request.py:62]: done

File: protostar/starknet_gateway/gateway_response_test.py (hunk 1 above)
Comment (Radinyn) [reply protostar/starknet_gateway/gateway_response_test.py:51]: done

**Review by mkaput — APPROVED** [review]

**Pull request merged**